src/tests/createHeadlessForm.test.js

```diff
@@ -1389,130 +1361,298 @@ describe('createHeadlessForm', () => {
+        await expect(validateForm(formValues)).toBeUndefined();
+
+        // when child.has_child is 'no' child.age is invisible
+        expect(formValues.child.age).toBe(null);
+      });
+
+      describe('supports conditionals to fieldsets', () => {
+        // To not mix the concepts:
+        // - Scoped conditionals: Conditionals written inside a fieldset
+        // - Conditionals to fieldsets: Root conditionals that affect a fieldset
+
+        // This describe has sequential tests, covering the following:
+        // If the working_hours > 30,
+        // Then the fieldset perks.food changes (the "no" option gets removed)
 
-      await expect(validateForm(formValues)).toBeUndefined();
-      expect(formValues.child.age).toBe(15);
+        // Setup (arrange)
+        const { fields, handleValidation } = createHeadlessForm(schemaWithConditionalToFieldset);
 
-      formValues.child.has_child = 'no';
-      // form value updates re-validate; see computeYupSchema()
-      await expect(validateForm(formValues)).toBeUndefined();
+        const validateForm = (vals) => friendlyError(handleValidation(vals));
+        const originalFood = getField(fields, 'perks', 'food');
+
+        const perksForLowWorkHours = {
+          food: 'no', // this option will be removed when the condition happens.
+          retirement: 'basic',
+        };
+
+        it('by default, the Perks.food has 4 options', () => {
+          expect(originalFood.options).toHaveLength(4);
+          expect(originalFood.description).toBeUndefined();
+
+          // Ensure the perks are required
+          expect(validateForm({})).toEqual({
+            work_hours_per_week: 'Required field',
+            perks: {
+              food: 'Required field',
+              retirement: 'Required field',
+            },
+          });
+
+          // Given low work hours, the form is valid.
+          expect(
+            validateForm({
+              work_hours_per_week: 5,
+              perks: perksForLowWorkHours,
+            })
+          ).toBeUndefined();
+        });
+
+        it('Given a lot work hours, the perks.food options change', () => {
+          expect(
+            validateForm({
+              work_hours_per_week: 35,
+            })
+          ).toEqual({
+            pto: 'Required field', // Sanity-check - this field gets required too.
+            perks: {
+              food: 'Required field',
+              retirement: 'Required field',
+            },
+          });
 
-      // when child.has_child is 'no' child.age is invisible
-      expect(formValues.child.age).toBe(null);
+          // The fieldset changed!
+          const foodField = getField(fields, 'perks', 'food');
+          // perks.food options changed ("No" was removed)
+          expect(foodField.options).toHaveLength(3);
+
+          // Ensure the "no" option is no longer accepted:
+          // This is a very important test in case the UI fails for some reason.
```

Comment:
note: This assertion is the reason why this branch is pointing to #18. A gif is worth 1000 words.

![Jul-02-2023 19-01-50](https://github.com/remoteoss/json-schema-form/assets/14869087/56b8c89e-0812-4ae8-87f9-b7948e5a75ee)

P.S. Tip for Remoters: To quickly preview this branch in our Playground, go to `useCreateHeadlessForm.js` (our internal hook), and replace the first line:

```diff
- import { createHeadlessForm } from '@remoteoss/json-schema-form';
// This assumes this repo is in the same folder as our internal FE repo.
+ import { createHeadlessForm } from '@/src/../../json-schema-form/src';
```